src/Mvc/Mvc.Core/src/Infrastructure/DefaultProblemDetailsFactory.cs

```diff
@@ -92,12 +92,18 @@ private void ApplyProblemDetailsDefaults(HttpContext httpContext, ProblemDetails
             problemDetails.Type ??= clientErrorData.Link;
         }
 
-        var traceId = Activity.Current?.Id ?? httpContext?.TraceIdentifier;
+        var traceId = Activity.Current?.TraceId.ToString() ?? httpContext?.TraceIdentifier;
         if (traceId != null)
         {
             problemDetails.Extensions["traceId"] = traceId;
         }
 
+        var traceParent = Activity.Current?.Id ?? httpContext?.TraceIdentifier;
```

Comment:
This seems wrong, I think it should be Activity.Current.ParentID or Activity.Current.ParentSpanID depending on the desired format - they are the same thing, just different representations.

And there probably should be a spanID to represent the current activity.